### PR TITLE
Add network module for NILinuxRT: nilrt_ip.py

### PR DESCRIPTION
**Pull request by heghe**
### What does this PR do?
Add network module for NILinuxRT

Connman is the network manager used as backend.
Functionalities:
  - ip.get_interface_details()
  - ip.up(interface)
  - ip.down(interface)
  - ip.set_dhcp_linklocal_all(interface)
  - ip.set_static_all(interface, address, netmask, gateway, domains)
  - ip.build_interface(interface, interface_type, enable, **settings)
  - ip.build_network_settings(**settings)
  - ip.get_network_settings()
  - ip.apply_network_settings(**settings)

Integration test for:
  - ip.enable
  - ip.disable
  - ip.set_dhcp_linklocal_all
  - ip.set_static_all

Signed-off-by: Heghedus Razvan <razvan.heghedus@ni.com>

### Tests written?

Yes

### CODE REVIEW

**Comment by cachedout** [comment]
Hi @raazvvann 

Thanks for this. There are a few issues that the linter picked up. Could you please take a look and make the requested changes? https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/7855/violations/

File: salt/modules/nilrt_ip.py [hunk 1]
```diff
@@ -0,0 +1,571 @@
...
+import time
+
+
+log = logging.getLogger(__name__)
+
+# Import third party libs
+try:
+    import pyconnman
+    import dbus
+except ImportError:
```
Comment (cachedout) [inline salt/modules/nilrt_ip.py]: What we probably want to do here is to move this into `__virtual__` and have that function return `False` if the `ImportError` is caught. That will ensure that Salt's loader system correctly displays the right error to the user and won't load the module when requested. There are many examples of this in various Salt execution modules.

**Review by cachedout — CHANGES_REQUESTED** [review]

**Comment by raazvvann** [comment]
I need to do some more work on this module, beside what you requested. I will let you know when is ready.

**Comment by raazvvann** [comment]
@cachedout I have done some rework on this module, please take a look and let me know if something isn't right

File: salt/modules/nilrt_ip.py [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+SERVICE_PATH = '/net/connman/service/'
+_CONFIG_TRUE = ['yes', 'on', 'true', '1', True]
+
+
+def __virtual__():
+    '''
+    Confine this module to NI Linux Real-Time based distros
+    '''
+    if __grains__['os_family'] == 'NILinuxRT':
+        if _get_state() == 'offline':
```
Comment (cachedout) [inline salt/modules/nilrt_ip.py]: Because `_get_state()` can raise an exception, we need to catch that here and then handle it appropriately instead of letting it bubble up.

File: salt/modules/nilrt_ip.py [hunk 3]
```diff
@@ -0,0 +1,515 @@
...
+
+
+def _connected(service):
+    '''
+    Verify if a connman service is connected
+    '''
+    state = pyconnman.ConnService(_add_path(service)).get_property('State')
+    if state == 'online' or state == 'ready':
+        return True
+    return False
```
Comment (cachedout) [inline salt/modules/nilrt_ip.py]: We could shorten this to `return state == 'online' or state == 'ready'` which will have the same result and be a little easier to read.

File: salt/modules/nilrt_ip.py (hunk 2 above)
Comment (raazvvann) [inline salt/modules/nilrt_ip.py]: Done

File: salt/modules/nilrt_ip.py (hunk 3 above)
Comment (raazvvann) [inline salt/modules/nilrt_ip.py]: Done
